presto-main/src/main/java/com/facebook/presto/cost/CoefficientBasedStatsCalculator.java

```diff
@@ -195,12 +180,12 @@ private PlanNodeStatsEstimate visitTableScanWithPredicate(TableScanNode node, Ex
             return tableScanStats;
         }
 
-        private Constraint<ColumnHandle> getConstraint(TableScanNode node, Expression predicate)
+        private Constraint<ColumnHandle> getConstraint(TableScanNode node)
         {
             DomainTranslator.ExtractionResult decomposedPredicate = DomainTranslator.fromPredicate(
```

Comment:
Is the decomposedPredicate still needed if the predicate is always TRUE_LITERAL?  Seems like this whole function can be replaced with `node.getCurrentConstraint`.

Reply:
Theoretically there might be the case that not the whole `node.getCurrentConstraint()` is used (see `decomposedPredicate.getRemainingExpression()`). I can't imagine this ever being the case (normally, we wouldn't store such a constraint in the `TableScanNode` in the first place), but I wouldn't hard-code this assumption here.

Reply:
I don't see how this helps:
`decomposedPredicate` will be an `ExtractionResult` of `TupleDomain.all()`.
`decomposedPredicate.getTupleDomain()` will give you `TupleDomain.all()`
`transform` does nothing because the domains was already an empty map
and then `intersect(node.getCurrentConstraint()` will just give you the `currentConstraint`.  

Am I missing something?

Reply:
I finally got that. You're right.